libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -689,21 +691,80 @@ bool experimental_table_slice_builder::add_impl(data_view x) {
 
 // -- utility functions --------------------------------------------------------
 
+namespace {
+
+std::vector<struct type::attribute>
+deserialize_attributes(std::string_view serialized) {
+  simdjson::padded_string json{serialized};
+  simdjson::dom::parser parser;
+  auto doc = parser.parse(json);
+  std::vector<struct type::attribute> attrs{};
+  for (auto f : doc.get_object()) {
+    auto value = std::string{f.value.get_string().value()};
+    // if (!f.value.is_string())
+    //   die("hard");
+    std::optional<std::string> value_opt
+      = value.empty() ? std::optional<std::string>() : value;
+    attrs.push_back({std::string{f.key}, value_opt});
+  }
+  return attrs;
+}
+
+std::string
+serialize_attributes(const std::vector<type::attribute_view>& attrs) {
+  auto out = std::string{};
+  auto inserter = std::back_inserter(out);
+  fmt::format_to(inserter, "{{ ");
+  bool first = true;
+  for (const auto& [key, value] : attrs) {
+    if (first)
+      first = false;
+    else
+      fmt::format_to(inserter, ", ");
+    fmt::format_to(inserter, R"("{}": "{}")", key, value);
+  }
+  fmt::format_to(inserter, "}}");
+  return out;
+}
+
+std::shared_ptr<const arrow::KeyValueMetadata>
+make_arrow_metadata(const type& t) {
+  auto metadata = std::make_shared<arrow::KeyValueMetadata>();
+  int nesting_level = 0;
+  for (const auto& [name, attrs] : t.names_and_attributes()) {
+    if (!name.empty())
+      metadata->Append(fmt::format("VAST:name:{}", nesting_level),
+                       std::string{name});
+    if (!attrs.empty())
+      metadata->Append(fmt::format("VAST:attributes:{}", nesting_level),
+                       serialize_attributes(attrs));
+    ++nesting_level;
+  }
+  if (nesting_level == 0)
+    return nullptr;
+  metadata->Append("VAST:max_level", fmt::format("{}", nesting_level - 1));
```

Comment:
I propose renaming this to `VAST:nesting-depth`, since nesting depth is both the term used for recursion limits inside both Arrow and VAST.

Not sure on the underscore vs. dashes situation, but we usually use dashes inside config keys.

Reply:
renamed to `VAST:nesting-depth` and the associated variable to `nesting_depth`.